src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -65,29 +65,60 @@ private static bool UseLoggerMessageDefine(LoggerMethod lm)
 
             private void GenType(LoggerClass lc)
             {
+                int indentationSize = 0;
+                string nestedIndentation = "";
                 if (!string.IsNullOrWhiteSpace(lc.Namespace))
                 {
                     _builder.Append($@"
 namespace {lc.Namespace}
 {{");
                 }
 
+                LoggerClass parent = lc.ParentClass;
+                Stack<string> parentClasses = new Stack<string>();
+                // loop until you find top level nested class
+                while (parent != null)
+                {
+                    parentClasses.Push($@"partial class {parent?.Name + " " + parent?.Constraints}");
```

Comment:
`struct` / `record`?

Reply:
log methods in structs are already not supported by the logging source generator. maybe it can be covered later.

e.g. with:

```c#

    internal partial struct MyStruct
    {
        private ILogger _logger;
        
        public MyStruct(ILogger logger) { _logger = logger; }

        [LoggerMessage(EventId = 9, Level = LogLevel.Debug, Message = "M9")]
        public partial void M9();
    }
```

we get: 
```
 error CS8795: Partial method 'NestedClass.M9()' must have an implementation part because it has accessibility modifiers. 
```

Reply:
> log methods in structs

I wasn't referring to log methods in structs. But log methods in a nested class of a struct / record. ex

```C#
public struct Customer
{
    private static class Logger
    {
        [LoggerMessage(EventId = 9, Level = LogLevel.Debug, Message = "M9")]
        public static void M9();
    }
}
```

```C#
public record Customer(string Name, string Address)
{
    private static class Logger
    {
        [LoggerMessage(EventId = 9, Level = LogLevel.Debug, Message = "M9")]
        public static void M9();
    }
}
```

Reply:
Added above use cases as tests. Also added support for adding logger methods directly in structs